### PR TITLE
💄(category) add sub categories in detail and better categories list

### LINKED ISSUE #541: Add list of categories page
## Feature Request

**Is your feature request related to a problem or unsupported use case? Please describe.**
Categories are a taxonomy that needs to be displayed on the page `/en/categories/`.

**Describe the solution you'd like**
- [x] Add a template that displays the whole nested structure of categories. The first level is a meta category (=filter on the search page), the immediate children are top level categories.
- [x] Add basic styling

### PR DESCRIPTION
**Pull request by sveetch**
## Purpose

Currently category list template use the default child list template
which have no special layout and we don't have a way to navigate
through category tree.

Fixes #541 

## Proposal

This commit add a new category list template which use a new category
glimpse template, also category plugin template has been changed to
use the glimpse so we have an unique template part to represent a
Category. By the way blogpost detail has been fixed to use new
'is_empty_placeholder' template filter.

### CODE REVIEW

File: src/richie/apps/courses/templates/courses/cms/category_list.html [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+{% block content %}
+<div class="category-glimpse-list">
+  <h1 class="category-glimpse-list__title">{{ current_page.get_title }}</h1>
+
+  {% for page in current_page.get_child_pages %}
+      {% include "courses/cms/fragment_category_glimpse.html" with category=page %}
+  {% empty %}
+    <p class="category-glimpse category-glimpse--empty">
+    {% trans "No associated categories" %}
```
Comment (sampaccoud) [inline src/richie/apps/courses/templates/courses/cms/category_list.html]: Why `associated`? Just "no categories"?

File: src/richie/apps/courses/templates/courses/cms/fragment_category_glimpse.html [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+            {% show_placeholder "logo" category as logo %}
+            {% if logo %}
+                {{ logo }}
+            {% else %}
+                <div class="category-glimpse__logo__empty">
+                {% trans "Logo" %}
+                </div>
+            {% endif %}
+        </div>
+        <div class="category-glimpse__title">{{ category.get_title }}</div>
```
Comment (sveetch) [inline src/richie/apps/courses/templates/courses/cms/fragment_category_glimpse.html]: This should be a `<hX>` based on `header_level` variable with default to h3

File: src/richie/apps/courses/templates/courses/cms/category_detail.html [hunk 3]
```diff
@@ -2,7 +2,7 @@
 {% load cms_tags i18n %}
 
 {% block content %}{% spaceless %}
-{% with category=current_page.category header_level=2 blogposts=current_page.category.get_blogposts courses=current_page.category.get_courses %}
+{% with category=current_page.category header_level=2 blogposts=current_page.category.get_blogposts courses=current_page.category.get_courses children=current_page.get_child_pages %}
```
[the remark this answers is not in the document]
Comment (sveetch) [reply src/richie/apps/courses/templates/courses/cms/category_detail.html]: @sampaccoud Done with new model method `Category.get_children_categories` and a test.

**Comment by sveetch** [comment]
Also updated CHANGELOG with forgotten changes from previous PR for person and this one.

File: CHANGELOG.md [hunk 4]
```diff
@@ -8,10 +8,17 @@ Versioning](https://semver.org/spec/v2.0.0.html).
...
+### Added
+
+- Add styling for persons list page,
+- Add sub categories in category detail page.
+
 ## Changed
 
 - Simplify Richie settings and provide defaults for those unlikely to be
   customized (search, languages, plugins, general).
+- Change layout global background to darker grey,
```
Comment (sampaccoud) [inline CHANGELOG.md]: the above line should get a ","?

File: CHANGELOG.md [hunk 5]
```diff
@@ -8,10 +8,17 @@ Versioning](https://semver.org/spec/v2.0.0.html).
 
 ## [Unreleased]
 
+### Added
+
+- Add styling for persons list page,
+- Add sub categories in category detail page.
```
Comment (sampaccoud) [inline CHANGELOG.md]: it seems you also added a category list page?

File: src/richie/apps/courses/models/category.py [hunk 6]
```diff
@@ -117,6 +117,24 @@ def get_blogposts(self, language=None):
...
+        """
+        Return a query to get only the children pages that are Category
+        objects related to this category. A convenient method since
+        ``page.get_child_pages`` return every direct child page, no matter
+        it's a Category or not.
+        """
+        page = (
+            self.extended_object
+            if self.extended_object.publisher_is_draft
+            else self.draft_extension.extended_object
```
Comment (sampaccoud) [inline src/richie/apps/courses/models/category.py]: if you do this you will display the draft pages on the public page. I think you must just do:
```python
return self.extended_object.get_child_pages().filter(category__isnull=False)
```

Comment (sveetch) [reply src/richie/apps/courses/models/category.py]: Changed as proposed

File: tests/apps/courses/test_models_category.py [hunk 7]
```diff
@@ -145,3 +145,34 @@ def test_models_category_get_blogposts_several_languages(self):
...
+
+        parent_category = CategoryFactory(should_publish=True)
+        CategoryFactory(
+            page_parent=parent_category.extended_object, should_publish=True
+        )
+        CategoryFactory(
+            page_parent=parent_category.extended_object, should_publish=True
+        )
+
+        BlogPostFactory.create_batch(
```
Comment (sampaccoud) [inline tests/apps/courses/test_models_category.py]: I don't see the point of adding this blogpost for the test. 
What you should add are pages of another type below the parent category page to check that they are not returned by the get_children_categories page?

Comment (sveetch) [reply tests/apps/courses/test_models_category.py]: @sampaccoud this to ensure there is nothing else than Category object is returned

File: CHANGELOG.md [hunk 8]
```diff
@@ -8,13 +8,21 @@ Versioning](https://semver.org/spec/v2.0.0.html).
 
 ## [Unreleased]
 
-### Changed
+### Added
+
+- Add template and styling for persons list page,
```
Comment (sampaccoud) [inline CHANGELOG.md]: I don't see anything related to persons in this PR?

Comment (sveetch) [reply CHANGELOG.md]: As said before ( https://github.com/openfun/richie/pull/636#issuecomment-482394215 ), i added missing change log from already merged PR for person list.

**Comment by sveetch** [comment]
@sampaccoud ready to merge, your commit fix included.

**Pull request merged**